waspc/src/Wasp/Analyzer/Parser/Lexer.x

```diff
@@ -97,6 +98,17 @@ alexGetByte (_, [], (currChar:remainingSource)) = case encodeChar currChar of
 alexInputPrevChar :: AlexInput -> Char
 alexInputPrevChar (prevChar, _, _) = prevChar
 
+-- | Tokenizes the given string. 
+alexScanTokens :: String -> Maybe [Parser Token]
```

Comment:
Signature is a bit unusual.
One would probably expect `Parser (Maybe [Token])`, instead of what you have now.
Actually, why not just `Parser [Token]`? Parser already has capability to throw errors, so why use Maybe instead and even lose information? Yeah, I would consider modifying the signature to `Parser [Token]`, since it makes more sense taking into account the Parser monad. `lexer` below returns `Parser a`. This here should return `Parser [Token]`. You could even add support for errors to return proper error locations. And now when I am writing this, I wonder if we should use `lexer` here, to implement this function. Because lexer already knows how to report lexer errors nicely. It feels like we are being pushed toward that solution naturally.

Reply:
Additional argument -> we are really doing some kind of parsing at the end, it is just the simplest form of parsing, trivial one, which is collecting all the tokens in one list. We can look at it that way. Parser from Parser.hs does proper parsing, it constructs AST, but here we tokenized. We can still see it as a trivial parser, possibly. hm.

Reply:
> You could even add support for errors to return proper error locations. And now when I am writing this, I wonder if we should use lexer here, to implement this function. Because lexer already knows how to report lexer errors nicely. It feels like we are being pushed toward that solution naturally.

Yes, that's what it seemed like when I wrote [this comment](https://github.com/wasp-lang/wasp/pull/520#discussion_r833796698). It appears we should be able to use the existing lexer to parse this. I'll think about it.

Reply:
I think either we use `lexer` in `alexScanTokens`, or we do regex. But I think the first options should actually turn out to be pretty elegant.

Reply:
Additional thought: We could even not add anything to Lexer.x at all then! We can just have `tokenize` in `Tokenizer.hs` that directly calls `lexer`?

Reply:
Yes, exactly, we just need to come up with a correct `(Token -> Parser a)` function.